src/main/java/com/example/expense/tracker/service/CategoryService.java

```diff
@@ -0,0 +1,51 @@
+package com.example.expense.tracker.service;
+
+import com.example.expense.tracker.repository.CategoryRepository;
+import com.example.expense.tracker.model.Category;
+import com.example.expense.tracker.model.CategoryRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class CategoryService {
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    public Category createCategory(CategoryRequest categoryRequest) {
+        Category category = new Category(categoryRequest.getName());
+        return categoryRepository.save(category);
+    }
+
+    public Category updateCategory(String name, CategoryRequest categoryRequest) {
```

Comment:
No need to return updated Category. Make it as void method.

Reply:
Updated the updateCategory method to have a void return type